torch/csrc/utils/python_scalars.h

```diff
@@ -36,7 +36,8 @@ inline void store_scalar(void* data, at::ScalarType scalarType, PyObject* obj) {
       *(uint32_t*)data = unpackIntegral<uint32_t>(obj, "uint32");
       break;
     case at::kUInt64:
-      *(uint64_t*)data = unpackIntegral<uint64_t>(obj, "uint64");
+      // NB: This doesn't allow implicit conversion of float to int
```

Comment:
Why the generic unpackIntegral didn't work out here?

Reply:
unpackIntegral unpacks to int64_t which cannot represent the upper half of uint64_t. So the unpackIntegral will fail with an overflow error. You can unpack it directly into uint64_t though, which is what we do below.